BackupBrowser: Add action to download backup's WordPress version

Initsogar opened the pull request:
## Proposed Changes
* Add action to download backup's WordPress version
* Remove margin-top on file's info when clicking on the WordPress version
* Refactor Tracks event handler to trigger when downloading the WordPress version, and also database tables.

## Screenshot
<img width="653" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/58ea0ee6-49be-4099-921c-9e2a16acceba">


## Testing Instructions
* Start a Jetpack Cloud or Calypso Live branch
* Select one of your sites with a Jetpack VaultPress Backup plan.
* Navigate to VaultPress Backup page
* Pick one date with a backup available (use the calendar and click on any date marked with a dot).
* Click on the first item listed in the browser (`WordPress version X.X.X`). It will load and display a `Download file` button.
* Click `Download file` and it will prompt the download for that WordPress version.

### Regression test
This PR affects the way the `Download` button is rendered and also how the Tracks events are triggered for this functionality. It will now vary depending on the type of file (`archive`, `table`, `wordpress` and other files), so I will recommend to do a regression testing validating WordPress, database tables, a plugin or theme, and any other file (like images, HTML file, etc.).
* Validate it downloads the desired file
* Validate it triggers the `t.gif` call for Tracks passing the appropriate file_type. This is similar to #78964.
  * <img width="150" alt="image" src="https://github.com/Automattic/wp-calypso/assets/1488641/f7e157a6-c65c-49df-8b98-525d8eda6f71">

 

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [x] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [x] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

Initsogar committed:
Add action to download backup's WordPress version

Initsogar committed:
Add a comment for setting an active node path

rustamshaikh reviewed (commented):
## Notes from testing: 

Not sure why I am unable to download the WordPress archive, I tried to use both the generated links to Calypso/Jetpack Cloud. Downloading other files works!

**Edit**: I am able to download the WordPress archive when entering the URL directly in the browser. 

https://github.com/Automattic/wp-calypso/assets/13975226/5bf46e35-0ef0-4f63-9422-d17f2e0663f1

Initsogar committed:
Refactor download button and Tracks

Given that we have the URL for downloading WordPress when rendering the
file info card, we are now using the href attribute instead of the
onclick to handle the download.

This commit also includes a refactor of the Tracks tracking action to
also be triggered when downloading a database table and WordPress.

Initsogar committed:
Rename `type` to `fileType`

Initsogar committed:
Remove optional chaining operator

Initsogar commented:
> Not sure why I am unable to download the WordPress archive, I tried to use both the generated links to Calypso/Jetpack Cloud. Downloading other files works!

Thanks @rustamshaikh! It seems we were returning the WordPress URL in HTTP instead of HTTPS. The fix is now ready for review in the VP repo.

Also took the opportunity to switch the button to a `<a href download>` approach given that we have the URL when we load the file info card component and introduced some small refactor on how we Tracks the download event.

jwebbdev reviewed (approved):
Works well with the API change
